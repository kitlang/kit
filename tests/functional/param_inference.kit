struct MyStruct[T] {
    var m: Map[Int, T];
}

function main() {
    var m: Map = Map.new(16);
    m.put(16, "hello");
    var s = m[16];
    puts(s);

    var m2: Map;
    m2 = Map.new(4);
    m2.put(true, 1);
    m2.put(false, 2);
    var i = m2[true];
    printf("%i\n", i);

    var st = struct MyStruct {
        m: Map.new(4)
    };
    st.m.put(1, 5);
    printf("%i\n", st.m[1]);
}
