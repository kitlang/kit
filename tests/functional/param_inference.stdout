hello
1
5
